### PR TITLE
Add tests for InflectedRoute

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: tests/TestCase/Routing/Route/InflectedRouteTest.php [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        ]);
+        $this->assertFalse($result);
+
+        $route = new InflectedRoute('/admin/subscriptions/:action/*', [
+            'controller' => 'Subscribe', 'prefix' => 'admin'
+        ]);
+        $result = $route->match([
+            'controller' => 'Subscribe',
+            'prefix' => 'admin',
+            'action' => 'edit_admin_e',
```
Comment (markstory) [inline tests/TestCase/Routing/Route/InflectedRouteTest.php]: Shouldn't some of these actions be camelBacked to ensure inflection happens?

File: tests/TestCase/Routing/Route/InflectedRouteTest.php [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        $route = new InflectedRoute(
+            '/media/search/*',
+            ['controller' => 'Media', 'action' => 'search_it']
+        );
+        $result = $route->parse('/media/search');
+        $this->assertEquals('Media', $result['controller']);
+        $this->assertEquals('search_it', $result['action']);
+        $this->assertEquals([], $result['pass']);
+
+        $result = $route->parse('/media/search/tv_shows');
```
Comment (markstory) [inline tests/TestCase/Routing/Route/InflectedRouteTest.php]: Same here.

File: tests/TestCase/Routing/Route/InflectedRouteTest.php (hunk 1 above)
Comment (ADmad) [inline tests/TestCase/Routing/Route/InflectedRouteTest.php]: Nope, InflectedRoute doesn't inflect action name when parsing nor matching.

Comment (markstory) [inline tests/TestCase/Routing/Route/InflectedRouteTest.php]: Ok.

**Comment by dereuromark** [comment]
'action' => 'my_view'

Maybe the docs should also point out more clearly that you also need the old `my_view` actions (method names) in controllers for this.
Quite a few people tripped over this.

We cannot just fix it to camelBacked due to BC.
But for all those that want to use PSR camelBacked actions and need to keep the underscore routing due to SEO reasons and CO, there is a fixed version available @ Shim plugin:
https://github.com/dereuromark/cakephp-shim/blob/master/docs/Features.md#route

**Pull request merged**